File: src/code-samples/deepagents/profiles-registration.ts
// :snippet-start: profiles-registration-js
import { registerHarnessProfile } from "deepagents";

// Set defaults for a hypothetical provider.
registerHarnessProfile("my_provider", {
  excludedTools: ["execute"],
  systemPromptSuffix: "Respond in under 500 words.",
});

// Override the prompt suffix for one model; inherit the excluded tool.
registerHarnessProfile("my_provider:my-model", {
  systemPromptSuffix: "Respond in under 100 words.",
});
// :snippet-end:

// :remove-start:
import assert from "node:assert/strict";
import { getHarnessProfile } from "deepagents";

let profile = getHarnessProfile("my_provider:my-model");
assert.ok(profile);
assert.deepEqual(new Set(profile.excludedTools), new Set(["execute"]));
assert.equal(profile.systemPromptSuffix, "Respond in under 100 words.");
const other = getHarnessProfile("my_provider:another-model");
assert.ok(other);
assert.deepEqual(new Set(other.excludedTools), new Set(["execute"]));
assert.equal(other.systemPromptSuffix, "Respond in under 500 words.");
// :remove-end:

// :snippet-start: profiles-reregister-harness-js
registerHarnessProfile("my_provider:my-model", {
  excludedTools: ["grep"],
});
// :snippet-end:

// :remove-start:
profile = getHarnessProfile("my_provider:my-model");
assert.ok(profile);
assert.deepEqual(new Set(profile.excludedTools), new Set(["execute", "grep"]));
assert.equal(profile.systemPromptSuffix, "Respond in under 100 words.");
assert.deepEqual(getHarnessProfile("my_provider:another-model"), other);
console.log("Profile registration and re-registration examples validated");
// :remove-end:

File: src/code-samples/deepagents/profiles-registration.py
"""Profiles: inherit provider defaults and extend a model registration."""

# :snippet-start: profiles-registration-py
from deepagents import HarnessProfile, register_harness_profile

# Set defaults for a hypothetical provider.
register_harness_profile(
    "my_provider",
    HarnessProfile(
        excluded_tools=frozenset({"execute"}),
        system_prompt_suffix="Respond in under 500 words.",
    ),
)

# Override the prompt suffix for one model; inherit the excluded tool.
register_harness_profile(
    "my_provider:my-model:tag",
    HarnessProfile(system_prompt_suffix="Respond in under 100 words."),
)
# :snippet-end:

# :remove-start:
from deepagents.profiles.harness.harness_profiles import _get_harness_profile

profile = _get_harness_profile("my_provider:my-model:tag")
assert profile is not None
assert profile.excluded_tools == frozenset({"execute"})
assert profile.system_prompt_suffix == "Respond in under 100 words."
other = _get_harness_profile("my_provider:another-model")
assert other is not None
assert other.excluded_tools == frozenset({"execute"})
assert other.system_prompt_suffix == "Respond in under 500 words."
# :remove-end:

# :snippet-start: profiles-reregister-harness-py
register_harness_profile(
    "my_provider:my-model:tag",
    HarnessProfile(excluded_tools=frozenset({"grep"})),
)
# :snippet-end:

# :remove-start:
profile = _get_harness_profile("my_provider:my-model:tag")
assert profile is not None
assert profile.excluded_tools == frozenset({"execute", "grep"})
assert profile.system_prompt_suffix == "Respond in under 100 words."
assert _get_harness_profile("my_provider:another-model") == other
print("Profile registration and re-registration examples validated")
# :remove-end:
